bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/ItemConsoleCommandExtension.java

```diff
@@ -78,6 +99,42 @@ public void execute(String[] args, Console console) {
         }
     }
 
+    private void removeTag(String itemName, String tag, Console console) {
+        Item item = itemRegistry.get(itemName);
+        if (item instanceof GenericItem) {
+            GenericItem gItem = (GenericItem) item;
+            gItem.removeTag(tag);
+            Item oldItem = itemRegistry.update(gItem);
+            if (oldItem == null) {
+                console.println("Error: Cannot remove tag " + tag + " from item " + itemName
+                        + " because this item does not belong to a ManagedProvider");
+            } else {
+                console.println("Successfully removed tag " + tag + " from item " + itemName);
+            }
+        } else {
+            console.println(
+                    "Error: Cannot remove tag " + tag + " from item " + itemName + " because it is not a GenericItem");
+        }
+    }
+
+    private void addTag(String itemName, String tag, Console console) {
+        Item item = itemRegistry.get(itemName);
+        if (item instanceof GenericItem) {
+            GenericItem gItem = (GenericItem) item;
+            gItem.addTag(tag);
```

Comment:
You don't need to use a function interface, you can also use a method directly as an argument to point the functional interface.

Let's me add an example (for static and non-static functions):

```java
import java.util.function.Function;

public class Test {

    public <T, R> R executeFunction(final Function<T, R> func, final T argument) {
        return func.apply(argument);
    }

    public String stringifyInteger(final Integer number) {
        return String.valueOf(number);
    }

    public String stringifyDouble(final Double number) {
        return String.valueOf(number);
    }

    public static String stringifyLong(final Long number) {
        return String.valueOf(number);
    }

    public static void main(final String[] args) {
        final Test test = new Test();

        final String stringifiedInteger = test.executeFunction(test::stringifyInteger, 1);
        final String stringifiedDouble = test.executeFunction(test::stringifyDouble, 1.2);
        final String stringifiedLong = test.executeFunction(Test::stringifyLong, 3L);
    }

}
```